blenderproc/python/types/URDFUtility.py

```diff
@@ -0,0 +1,276 @@
+            for obj in link.get_children():
+                obj.set_cp(key="category_id", value=category_id)
+
+    def remove_link_by_index(self, index: int = 0):
+        """ Removes a link and all its associated objects given an index. Also handles relationship of the link's child with its parent.
+        This is useful for removing a 'world link' which could be a simple flat surface, or if someone wants to shorten the whole urdf object.
+
+        :param index: Index of the joint to be removed.
+        """
+        assert index < len(self.links), f"Invalid index {index}. Index must be in range 0, {len(self.links)} (no. links: {len(self.links)})."
+
+        # remove link from the urdf instance and determine child / parent
+        parent = self.links[index - 1] if index != 0 else None
+        child = self.links[index + 1] if index != len(self.links) else None
+        link_to_be_removed = self.links.pop(index)
+
+        # calculate new transformation of child and set new parent
+        if child is not None:
+            child.blender_obj.matrix_world = link_to_be_removed.blender_obj.matrix_world
+            bpy.ops.object.transform_apply(location=True, rotation=True, scale=True, properties=True)
+
+            # handle new parent-child relationship
+            if parent is not None:
+                bpy.ops.object.select_all(action='DESELECT')
+                parent.select()
+                child.select()
+
+                # select which object to parent to
+                bpy.context.view_layer.objects.active = parent.blender_obj
+
+                # parent object
+                bpy.ops.object.posemode_toggle()
+                bpy.ops.object.parent_set(type="BONE")
+                bpy.ops.object.posemode_toggle()
+
+        # set new base in case the original base will be removed
+        if index == 0:
+            super().__init__(bpy_object=self.links[0].blender_obj)
+
+        # finally, delete the link and all its children
+        for obj in link_to_be_removed.get_children():
+            obj.delete()
+        link_to_be_removed.delete()
+
+    def set_name(self, name: str):
+        """ Sets the name of the urdf object.
+
+        :param name: The new name.
+        """
+        self.name = name
+
+    def get_name(self) -> str:
+        """ Returns the name of the urdf object.
+
+        :return: The name.
+        """
+        return self.name
+
+    def hide(self, hide_object: bool = True):
+        """ Sets the visibility of the object.
+
+        :param hide_object: Determines whether the object should be hidden in rendering.
+        """
+        for link in self.links:
+            link.hide(hide_object=hide_object)
+            for child in link.get_children():
+                child.hide(hide_object=hide_object)
+
+    def get_all_local2world_mats(self):
+        """ Returns all matrix_world matrices from every joint. """
+
+        return np.stack([link.blender_obj.matrix_world for link in self.links])
```

Comment:
```suggestion
        return np.stack([link.get_local2world_mat() for link in self.links])
```
`matrix_world` is sometimes not updated